kmk/kmktime.py

```diff
@@ -1,23 +1,22 @@
+import supervisor
+
+import const
 import time
 
+_TICKS_PERIOD = const(1 << 29)
+_TICKS_MAX = const(_TICKS_PERIOD - 1)
+_TICKS_HALFPERIOD = const(_TICKS_PERIOD // 2)
+
 
 def sleep_ms(ms):
     return time.sleep(ms / 1000)
 
 
 def ticks_ms():
-    '''Has .25s granularity, but is cheap'''
-    return time.monotonic() * 1000
+    return supervisor.ticks_ms() * 1000
 
 
 def ticks_diff(new, old):
-    return new - old
-
-
-def accurate_ticks():
-    '''Is more expensive, but good for time critical things'''
-    return time.monotonic_ns()
-
-
-def accurate_ticks_diff(new, old, ms):
```

Comment:
`ticks_diff` always returned a signed difference, but the way I read it `accurate_ticks_diff` returned a bool indicating before/after a deadline (and needs 3 args, not two). The way you're using it below, it looks like you expect a boolean result.

I'd have explained the old `accurate_ticks_diff` as: Return `True` iff the difference between `new` and `old` is less than `ms`. This becomes something like
```
def check_deadline(new, old, ms):
    """Return True iff less than `ms` milliseconds have passed between `old` and `new`"""
    return ticks_diff(now, start) < ms
```

Reply:
Added this. Much appreciated 👍